### PR TITLE
Make the integration tests compatible with the dev environments

### PR DESCRIPTION
**Pull request by robinjam**
[Trello card](https://trello.com/c/D1m5uleD/952-run-api-client-integration-tests-as-part-of-new-pipeline)

Some small updates to make the integration tests runnable in the dev environments via the new deploy pipeline.

Also updates the integration tests to print a stack trace when they fail, to make debugging easier.

[Tested in dev-a](https://concourse.notify.tools/teams/dev-a/pipelines/deploy-notify/jobs/test/builds/162) (5 times in a row to check for flakiness)

### CODE REVIEW

File: bin/test_client.rb [hunk 1]
```diff
@@ -65,12 +62,10 @@ def test_get_all_templates(client)
...
   response = client.get_all_templates('type' => 'sms')
   unless response.is_a?(Notifications::Client::TemplateCollection)
-    p 'failed test_get_all_templates response is not a Notifications::Client::TemplateCollection'
-    exit 1
+    raise 'failed test_get_all_templates response is not a Notifications::Client::TemplateCollection'
   end
   unless response.collection.length >= 1
-    p 'failed test_get_all_templates, expected at least 2 templates returned.'
-    exit 1
+    raise 'failed test_get_all_templates, expected at least 2 templates returned.'
```
Comment (risicle) [inline bin/test_client.rb:68]: > at least 2
at least 1?

**Review by risicle — APPROVED** [review]
LGTM apart from typo (admittedly not your typo)

**Comment by risicle** [comment]
Do you not need to add a step to the dockerfile to copy the code though?

**Comment by robinjam** [comment]
> Do you not need to add a step to the dockerfile to copy the code though?

The dockerfile already has `COPY . .` prior to this commit (to allow the `bundle install` called by `make bootstrap` to work)

File: bin/test_client.rb (hunk 1 above)
Comment (robinjam) [reply bin/test_client.rb]: Good catch, fixed! (Will need a re-review)

**Comment by risicle** [comment]
Ohh didn't see it unfolds downwards

**Review by risicle — APPROVED** [review]

**Pull request merged**